features/flows/elastic/src/main/java/org/opennms/netmgt/flows/elastic/Direction.java

```diff
@@ -28,23 +28,27 @@
 
 package org.opennms.netmgt.flows.elastic;
 
-import java.util.Objects;
+import org.opennms.netmgt.flows.api.Flow;
 
 import com.google.gson.annotations.SerializedName;
 
 public enum Direction {
     @SerializedName("ingress")
-    INGRESS("ingress"),
+    INGRESS,
     @SerializedName("egress")
-    EGRESS("egress");
+    EGRESS;
 
-    private final String value;
+    public static Direction from(Flow.Direction direction) {
+        if (direction == null) return null;
 
-    private Direction(String value) {
-        this.value = Objects.requireNonNull(value);
+        switch (direction) {
+            case INGRESS:
+                return INGRESS;
+            case EGRESS:
+                return EGRESS;
+            default:
+                throw new IllegalArgumentException();
```

Comment:
Can we add some meaning to what actually went wrong? `throw new IllegalArgumentException("Could not map <input> to a valid Direction.`. In addition we should add a test - if not already added - which verifies that each `Flow.Direction` can be converted to a `Direction`.